Covenant Training Pistol

Boltersam opened the pull request:
adds Covenant training pistol. It's just a recoloured plasma pistol with TTR coded projectiles, for training purposes.

Commit:
Update ammo.dm

Commit:
Add files via upload

Commit:
Update pistols.dm

caelaislinn commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -60,6 +60,55 @@
+
+/obj/item/weapon/gun/energy/trainingpistol/disabled/attack_self(var/mob/user)
+	desc = "A dual funtionality pistol: It fires bolts of plasma, and when overcharged is capable of emitting a small emp burst at the point of impact. This one appears to be modified to fire very weak bolts of energy. It is disabled"
+	max_shots = 0
```
Don't bother with this one, the /disabled cov weapons are purely show pieces for the research base

caelaislinn commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -60,6 +60,55 @@
+	var/overcharge_cost = 1
+
+
+/obj/item/weapon/gun/energy/trainingpistol
```
This should have `/New()` on the end of the line

caelaislinn commented on code/modules/halo/weapons/covenant/pistols.dm:
```diff
@@ -60,6 +60,55 @@
+	if(new_overcharge != overcharge)
+
+
+/obj/item/weapon/gun/energy/trainingpistol/disabled/attack_self(var/mob/user)
```
Remove `/attack_self(var/mob/user)` from the end of the line

caelaislinn commented:
Pull request needs to be renamed to "Covenant training pistol" or some such

Commit:
Update pistols.dm

fixed training pistol

Commit:
Update pistols.dm

Commit:
Update pistols.dm

Commit:
Update pistols.dm

Commit:
Update pistols.dm

Commit:
Deletes unnecessary lines, adds thing to line 91

Deletes Lines 78, 88, 99,106
Line 39 unnecessary New() removed
Line 91 gets /cov_plasma_recharge_tick()
as recommended by XO
